### PR TITLE
Pick changes from PR 33811, use UID in dashboard navlinks

### PR DESCRIPTION
**Pull request by zserge**
As the original PR #33811 stagnated, and we would like to proceed with merging the suggested changes - here's a manual cherry-pick of that PR with the unused method being removed.

### CODE REVIEW

File: pkg/plugins/models.go [hunk 1]
```diff
@@ -95,10 +95,19 @@ type PluginInclude struct {
 	DefaultNav bool            `json:"defaultNav"`
 	Slug       string          `json:"slug"`
 	Icon       string          `json:"icon"`
+	Uid        string          `json:"uid"`
```
Comment (wbrowne) [inline pkg/plugins/models.go:98]: We should add this new field to the JSON schema https://github.com/grafana/grafana/blob/main/docs/sources/developers/plugins/plugin.schema.json#L58

**Review by wbrowne — COMMENTED** [review]
LGTM. Tested a bit and it does what I expect it to do. Would be good to add the new field to the schema is all I think

File: pkg/plugins/models.go (hunk 1 above)
Comment (wbrowne) [reply pkg/plugins/models.go:98]: Nit:
```suggestion
	UID        string          `json:"uid"`
```

File: docs/sources/developers/plugins/plugin.schema.json [hunk 2]
```diff
@@ -62,6 +62,10 @@
         "type": "object",
         "additionalItems": false,
         "properties": {
+          "uid": {
+            "type": "string",
+            "description": "Unique identifier of the included resouce"
```
Comment (achatterjee-grafana) [inline docs/sources/developers/plugins/plugin.schema.json:67]:
```suggestion
            "description": "Unique identifier of the included resource"
```

**Review by achatterjee-grafana — COMMENTED** [review]
Fixed a typo.

**Review by idafurjes — APPROVED** [review]
LGTM

**Review by achatterjee-grafana — APPROVED** [review]

**Review by wbrowne — APPROVED** [review]
Looks good apart from typo!

**Pull request merged**